src/NUnitFramework/framework/Constraints/EqualConstraintResult.cs

```diff
@@ -241,6 +241,19 @@ private static object GetValueFromCollection(ICollection collection, int index)
 
             return null;
         }
+
+        private static IEnumerable SkipItems(IEnumerable enumerable, long skip)
+        {
+            var iterator = enumerable.GetEnumerator();
+            while (skip-- > 0)
+            {
+                iterator.MoveNext();
```

Comment:
`.Cast<object>().Skip(...)` is simple and available here on all target platforms. But if we go ahead with this, I think it's contractually invalid to call `MoveNext` if it returns false, so we should yield break. And maybe move it to an internal extension methods class.

Reply:
And *technically* we should be doing `using (iterator as IDisposable)` around it too, to match the behavior of `foreach`.

Reply:
1) Skip is not available, as it doesn't accept long argument (failurePoint.Position).
(just curious - do we actually support collections with more than int.MaxValue elements?)

2) While I think it makes sense to add `yield break` (or some other if condition) to avoid unneeded calls, could you suggest where could I read that it's invalid? That sounds strange to me a bit (since in that case I cannot determine lookng at enumerator object whether I'm allowed to call MoveNext)?

3) Could you suggest where could I put it? Or create new class?

Reply:
1. Oh my, I suppose we would support enumerables longer than int.MaxValue!

2. I just checked myself; I was wrong. Calling `MoveNext` again after the last call returned false [is actually legal](https://docs.microsoft.com/dotnet/api/system.collections.ienumerator.movenext#remarks). (I was thinking of Current, which [should not be called](https://docs.microsoft.com/dotnet/api/system.collections.generic.ienumerator-1.current) after MoveNext returns false.)
Either way, it doesn't make sense to me to ignore the return value of MoveNext.

3. Probably the `NUnit.Framework.Extensions` class.

Reply:
Updated.